Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBService/ChangeStreamSession.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import StitchCoreSDK
+import StitchCoreRemoteMongoDBService
+
+/**
+ * A MongoDB Change Stream session. This can be used to get or change the delegate that reacts to change events on
+ * this stream. It can also be used to close open streams.
+ */
+public class ChangeStreamSession<DelegateT: ChangeStreamDelegate> {
+    /**
+     * The delegate for this `ChangeStreamSession`. This is the class that will react to incoming events. A delegate is
+     * set at the time the `watch` function is called, but a new delegate may also be set here. This delegate type
+     * must conform to `ChangeStreamDelegate`
+     *
+     * - warning the delegate is held by the change stream as a weak reference. If the delegate becomes
+     *           deallocated, the stream will automatically close the next time an event is received. If you are
+     *           setting a new delegate, make sure you do so before the previous delegate is deallocated.
+     */
+    public var delegate: DelegateT? {
+        get {
+            return self.internalDelegate.delegate
+        }
+        set(newValue) {
```

Comment:
what would happen if you set the delegate to nil?

Reply:
the next time the stream got an event, it would close the stream. 

this is optional because the underlying stream holds a weak var.

check out https://blog.bobthedeveloper.io/swift-retention-cycle-in-closures-and-delegate-836c469ef128 for why we hold the delegate weakly